### DISCUSSION TITLE
How to combine suggested ssh alias with GPG for SSH with passphrase

### DISCUSSION THREAD

**Discussion by hedefalk** [kovidgoyal/kitty#3337] [Q&A]
So I've added the following to my fish login setup to help with ssh:ing into servers that hasn't kitty's terminfo as per faq: 
https://sw.kovidgoyal.net/kitty/faq.html#i-get-errors-about-the-terminal-being-unknown-or-opening-the-terminal-failing-when-sshing-into-a-different-computer

```
# alias ssh in kitty
if test "$TERM" = "xterm-kitty"
  alias ssh="kitty +kitten ssh"
end
```

However, I also want to be able to use GPG for SSH instead of keys under ~/.ssh. This works fine under other terminal editors and also if I remove the alias:

```
# Make SSH use GPG
set -e SSH_AGENT_PID
set -x SSH_AUTH_SOCK (gpgconf --list-dirs agent-ssh-socket)
set -x GPG_TTY (tty)
gpg-connect-agent updatestartuptty /bye >/dev/null
```

I guess the `GPG_TTY(tty)` construct breaks because of this alias and GPG tries to open up a passphrase dialog in the wrong terminal?

Like so:
```
❯ ssh -T git@github.com
Hi hedefalk! You've successfully authenticated, but GitHub does not provide shell access.

❯ alias ssh="kitty +kitten ssh"

❯ ssh -T git@github.com
PTY allocation request failed on channel 0
```

Any tips on how to manage this better?

**Comment by kovidgoyal** [comment]
You cant use -T with the ssh kitten.

**Comment by kovidgoyal** [comment]
Or rather it will be ignored since the ssh kitten adds -t when calling ssh

**Reply by hedefalk** [reply]
Ok, thanks! Behaviour is the same without though:

```
> ssh git@github.com
PTY allocation request failed on channel 0
```

**Reply by hedefalk** [reply]
Hm, I still get the PTY allocation error without the alias but it also works:

In kitty:
```
✗ ssh git@github.com
PTY allocation request failed on channel 0
 
✗ functions --erase ssh
 
❯ ssh git@github.com
PTY allocation request failed on channel 0
Hi hedefalk! You've successfully authenticated, but GitHub does not provide shell access.
Connection to github.com closed.
```

MacOs Terminal:
```
✗ ssh git@github.com
PTY allocation request failed on channel 0
Hi hedefalk! You've successfully authenticated, but GitHub does not provide shell access.
Connection to github.com closed.

✗ ssh -T git@github.com
Hi hedefalk! You've successfully authenticated, but GitHub does not provide shell access.
```

**Comment by kovidgoyal** [comment]
I'm confused if you are sshing into a server that does not provide shell access, why are you using the kitten at all??
